File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/ImageCardParameters.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// SharePoint image card parameters.
    /// </summary>
    public class ImageCardParameters : BaseCardParameters, ICardParameters
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ImageCardParameters"/> class.
        /// </summary>
        public ImageCardParameters()
        {
            // Do nothing
        }

        /// <summary>
        /// Gets or Sets the primary text of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the primary text to display.</value>
        [JsonProperty(PropertyName = "primaryText")]
        public string PrimaryText { get; set; }

        /// <summary>
        /// Gets or Sets the image url of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the image url of the card view.</value>
        [JsonProperty(PropertyName = "imageUrl")]
        public string ImageUrl { get; set; }

        /// <summary>
        /// Gets or Sets the image alt text of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the image alt text of the card view.</value>
        [JsonProperty(PropertyName = "imageAltText")]
        public string ImageAltText { get; set; }
    }
}

File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/imageCardParameters.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BaseCardParameters } from './baseCardParameters';

/**
 * Sharepoint image card view parameters
 */
export class ImageCardParameters extends BaseCardParameters {
    private primaryText: string;
    private imageUrl: string;
    private imageAltText: string;

    /**
     * Initializes a new instance of the ImageCardParameters class
     */
    public ImageCardParameters() {
        // Do nothing
    }

    /**
     * Sets title property of type string
     */
    public set PrimaryText(primaryText: string) {
        this.primaryText = primaryText;
    }

    /**
     * Gets title property of type string
     */
    public get PrimaryText(): string {
        return this.primaryText;
    }

    /**
     * Sets image url property of type string
     */
    public set ImageUrl(imageUrl: string) {
        this.imageUrl = imageUrl;
    }

    /**
     * Gets image url property of type string
     */
    public get ImageUrl(): string {
        return this.imageUrl;
    }

    /**
     * Sets image alt text property of type string
     */
    public set ImageAltText(imageAltText: string) {
        this.imageAltText = imageAltText;
    }

    /**
     * Gets image alt text property of type string
     */
    public get ImageAltText(): string {
        return this.imageAltText;
    }
}